install_files/ansible-base/roles/common/tasks/apt_upgrade.yml

```diff
@@ -40,31 +40,18 @@
   # will not install security patches.
   changed_when: tor_hold_package_result|changed
 
-- name: Check if reboot is required due to security updates.
-  stat:
-    path: /var/run/reboot-required
-  register: reboot_required
-  # Task is read-only, so don't mark changed.
-  changed_when: false
-  tags:
-    - reboot
 
-- name: Reboot if required due to security updates.
-  command: reboot
-  when: reboot_required.stat.exists
-  tags:
-    - reboot
-
-- name: Wait for server to come back.
-  local_action: wait_for
+  # Ansible doesn't support notifying handlers based on file existence, results,
+  # e.g. via registered `stat` results, only only on changed=true task results,
+  # so let's craft a task that returns changed=true if-and-only-if a file exists.
+- name: Check if reboot is required due to security updates.
```

Comment:
:laughing: this task is ansible hackery at it's finest. :champagne: no seriously, i like the work-around here - its sweet how adapt you've gotten at working around some of ansible's current limitations.